sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs

```diff
@@ -1057,5 +1063,16 @@ async Task<Response<BlobContentInfo>> UploadStreamAsync()
             }
         }
         #endregion Upload
+
+        /// <summary>
+        /// Performs a transform on the data for uploads. It is a no-op by default.
+        /// </summary>
+        /// <param name="content">Content to transform.</param>
+        /// <param name="metadata">Content metadata to transform.</param>
+        /// <returns>Transformed content stream and metadata.</returns>
+        internal protected virtual (Stream, Metadata) TransformContent(Stream content, Metadata metadata)
```

Comment:
Remove the `protected` - I'm not sure we want this to be a public API without more consideration.

Reply:
Oh! I had always thought `internal protected` was the intersection of that visibility, not the union. Will fix.